azure-pipeline-codeql.yml

```diff
@@ -0,0 +1,155 @@
+      - pwsh: |
+          . $(Build.SourcesDirectory)\eng\CodeQL.ps1
+          $guardianCliLocation = Install-Gdn -Path $(Build.SourcesDirectory)\.artifacts
+          Write-Host "##vso[task.setvariable variable=GuardianCliLocation]$guardianCliLocation"
+        displayName: Install Guardian
+
+      # Project requirements
+      - task: UseDotNet@2
+        displayName: Install Correct .NET Version
+        inputs:
+          useGlobalJson: true
+
+      - task: UseDotNet@2
+        displayName: Install .NET Version 3.1
+        inputs:
+          version: 3.1.x
+
+      - task: NodeTool@0
+        displayName: 'Install Node'
+        inputs:
+          versionSpec: '12.x'
+
+      - task: NuGetCommand@2
+        displayName: Restore Packages
+        inputs:
+          command: restore
+          solution: "**/*.sln"
+          feedstoUse: config
+
+      - pwsh: eng\set-version-parameters.ps1
+        displayName: Calculate release version variables
+        
+      - pwsh: |
+          [xml]$manifest = Get-Content src\Maestro\MaestroApplication\ApplicationPackageRoot\ApplicationManifest.xml
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Policies']").RemoveAll()
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Principals']").RemoveAll()
+          $manifest.Save("src\Maestro\MaestroApplication\ApplicationPackageRoot\ApplicationManifest.xml")
+          git diff
+        displayName: Remove Service Fabric RunAsPolicy from MaestroApplication
+
+      - pwsh: |
+          [xml]$manifest = Get-Content src\Telemetry\TelemetryApplication\ApplicationPackageRoot\ApplicationManifest.xml
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Policies']").RemoveAll()
+          $manifest.SelectSingleNode("/*[local-name()='ApplicationManifest']/*[local-name()='Principals']").RemoveAll()
+          $manifest.Save("src\Telemetry\TelemetryApplication\ApplicationPackageRoot\ApplicationManifest.xml")
+          git diff
+        displayName: Remove Service Fabric RunAsPolicy from TelemetryApplication
+      
+      - pwsh: |
+          . $(Build.SourcesDirectory)\eng\codeql.ps1
+          Initialize-Gdn -GuardianCliLocation $(GuardianCliLocation) `
+            -WorkingDirectory $(Build.SourcesDirectory) `
+            -LoggerLevel 'Standard'
+        displayName: 'CodeQL: Initialize'
+
+      - pwsh: |
+          . $(Build.SourcesDirectory)\eng\codeql.ps1
+          New-GdnSemmelConfig -GuardianCliLocation $(GuardianCliLocation) `
+            -LoggerLevel 'Standard' `
+            -Language 'csharp' `
+            -WorkingDirectory $(Build.SourcesDirectory) `
+            -SourceCodeDirectory $(Build.SourcesDirectory)\src `
+            -BuildCommand "$(Build.SourcesDirectory)\build.cmd -configuration Release -prepareMachine /p:Test=false /P:Sign=false # rmdir /s /q $(Build.SourcesDirectory)\src\Maestro\maestro-angular\dist" `
```

Comment:
What's the commented-out rmdir here?

Reply:
That's actually a line delimiter in this context. CodeQL will execute both lines in a batch file separately. 

The usual build process has a little path that generates javascript for the Maestro Angular app. CodeQL thinks it is regular code and blames me for its problems. CodeQL doesn't want generated code when it analyzes the Angular app, so I just delete it here.